scripts/datascience/themes_quality/wiki_puzzle_gemini_pipeline/mp_puzzle_generation/ThemeWorker.py

```
@@ -194,19 +194,27 @@ def _get_embeddings_with_cache(self, texts: List[str]) -> Tuple[List[Optional[Li
+                with self.rate_limiter.acquire_request_slot():
+                    api_embeddings = self.embedding_provider.generate_embeddings(texts_to_generate)
+                    api_calls = 1  # One batch API call
+                
+                # Merge API results back into cached results
+                api_idx = 0
+                for i in text_indices:
+                    if api_idx < len(api_embeddings) and api_embeddings[api_idx] is not None:
+                        cached_embeddings[i] = api_embeddings[api_idx]
+                        # Cache the new embedding
+                        self.cache_manager.put(texts[i], api_embeddings[api_idx])
+                    api_idx += 1
```

Ruff (0.12.2)

Use `enumerate()` for index variable `api_idx` in `for` loop

(SIM113)